probe-rs/src/rtt/channel.rs

```diff
@@ -50,27 +52,46 @@ impl Channel {
     const O_WRITE: usize = 12;
     const O_READ: usize = 16;
     const O_FLAGS: usize = 20;
+    const O_BUFFER_PTR_64: usize = 8;
+    const O_SIZE_64: usize = 16;
+    const O_WRITE_64: usize = 24;
+    const O_READ_64: usize = 32;
+    const O_FLAGS_64: usize = 40;
 
     pub(crate) fn from(
         core: &mut Core,
         number: usize,
         memory_map: &[MemoryRegion],
-        ptr: u32,
+        ptr: u64,
         mem: &[u8],
+        is_64bit: bool,
     ) -> Result<Option<Channel>, Error> {
-        let buffer_ptr: u32 = match mem.pread_with(Self::O_BUFFER_PTR, LE) {
-            Ok(buffer_ptr) => buffer_ptr,
-            Err(_error) => return Err(Error::MemoryRead("RTT channel address".to_string())),
+        let buffer_ptr: u64 = match if is_64bit {
+            let p: Result<u64, scroll::Error> = mem.pread_with(Self::O_BUFFER_PTR_64, LE);
+            p
+        } else {
+            let p: Result<u32, scroll::Error> = mem.pread_with(Self::O_BUFFER_PTR, LE);
+            p.map(u64::from)
+        } {
+            Ok(p) => p,
+            Err(_e) => return Err(super::Error::MemoryRead("RTT channel address".to_string())),
         };
 
         if buffer_ptr == 0 {
             // This buffer isn't in use
             return Ok(None);
         }
 
-        let name_ptr: u32 = match mem.pread_with(Self::O_NAME, LE) {
-            Ok(name_ptr) => name_ptr,
-            Err(_error) => return Err(Error::MemoryRead("RTT channel name".to_string())),
+        // TODO ここの仕組みを直したい
```

Comment:
👀

Reply:
Oops, I'm embarrassed.
remove it fc89d0b8eecaebb8b180da79aec796f56e22e884